deepchecks/tabular/utils/task_inference.py

```diff
@@ -39,65 +40,58 @@ def get_possible_classes(model: Optional[BasicModel], train_dataset: 'tabular.Da
         Train Dataset of task
     test_dataset : Optional['tabular.Dataset'], default = None
         Test Dataset of task
-    force_classification: bool, default = False
-        Whether to disable the auto infer for task type and return all observed label values.
+    model_classes
+        Model's classes if provided by the user manually.
+
     Returns
     -------
-    Optional[List]
-        The list of possible classes for classification tasks or None for regression
+    (TaskType, List, List)
+        The type of the Task, The observed classes, The model classes
     """
-    if not isinstance(train_dataset, tabular.Dataset) or (test_dataset is not None and
-                                                          not isinstance(test_dataset, tabular.Dataset)):
-        raise DeepchecksValueError('train_dataset and test_dataset must be of type tabular.Dataset')
+    train_labels = []
+    test_labels = []
+    have_model = model is not None
+    if train_dataset:
+        if train_dataset.has_label():
+            train_labels += train_dataset.label_col.to_list()
+        if have_model:
+            train_labels += convert_into_flat_list(model.predict(train_dataset.features_columns))
+    if test_dataset:
+        if test_dataset.has_label():
+            test_labels += test_dataset.label_col.to_list()
+        if have_model:
+            test_labels += convert_into_flat_list(model.predict(test_dataset.features_columns))
 
-    if train_dataset._label_classes is not None:
-        if hasattr(model, 'classes_') and len(model.classes_) > 0 and \
-                list(model.classes_) != train_dataset._label_classes:
-            raise DeepchecksValueError('Model output classes and train dataset label classes do not match')
-        return train_dataset._label_classes
+    observed_labels = pd.Series(test_labels + train_labels)
+    if model_classes is None and have_model and hasattr(model, 'classes_') and len(model.classes_) > 0:
+        model_classes = sorted(list(model.classes_))
 
-    observed_labels = list(train_dataset.label_col)
-    if test_dataset is not None:
-        observed_labels += list(test_dataset.label_col)
-    if hasattr(model, 'classes_') and len(model.classes_) > 0:
-        if not set(pd.Series(observed_labels).dropna().unique()).issubset(set(model.classes_)):
-            get_logger().warning('Model classes attribute does not contain all observed labels in train and test data.')
-        observed_labels += list(model.classes_)
+    if train_dataset and train_dataset.label_type is not None:
+        task_type = train_dataset.label_type
+    elif model_classes:
+        task_type = infer_by_class_number(len(model_classes))
+    elif len(observed_labels) > 0 and is_categorical(observed_labels, max_categorical_ratio=0.05):
```

Comment:
```suggestion
    elif is_categorical(observed_labels, max_categorical_ratio=0.05):
```

Reply:
That's not possible, since we can have checks with only model, and then `obersved_classes` is empty, and the is_categroical fails

Reply:
ah ok